tests/lib/rules/no-reserved-component-names.js

```diff
@@ -0,0 +1,324 @@
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-reserved-component-names')
+const RuleTester = require('eslint').RuleTester
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const invalidElements = [
+  'annotation-xml', 'AnnotationXml',
+  'color-profile', 'ColorProfile',
+  'font-face', 'FontFace',
+  'font-face-src', 'FontFaceSrc',
+  'font-face-uri', 'FontFaceUri',
+  'font-face-format', 'FontFaceFormat',
+  'font-face-name', 'FontFaceName',
+  'missing-glyph', 'MissingGlyph',
+  'html', 'Html',
+  'body', 'Body',
+  'base', 'Base',
+  'head', 'Head',
+  'link', 'Link',
+  'meta', 'Meta',
+  'style', 'Style',
+  'title', 'Title',
+  'address', 'Address',
+  'article', 'Article',
+  'aside', 'Aside',
+  'footer', 'Footer',
+  'header', 'Header',
+  'h1', 'H1',
+  'h2', 'H2',
+  'h3', 'H3',
+  'h4', 'H4',
+  'h5', 'H5',
+  'h6', 'H6',
+  'hgroup', 'Hgroup',
+  'nav', 'Nav',
+  'section', 'Section',
+  'div', 'Div',
+  'dd', 'Dd',
+  'dl', 'Dl',
+  'dt', 'Dt',
+  'figcaption', 'Figcaption',
+  'figure', 'Figure',
+  'hr', 'Hr',
+  'img', 'Img',
+  'li', 'Li',
+  'main', 'Main',
+  'ol', 'Ol',
+  'p', 'P',
+  'pre', 'Pre',
+  'ul', 'Ul',
+  'a', 'A',
+  'b', 'B',
+  'abbr', 'Abbr',
+  'bdi', 'Bdi',
+  'bdo', 'Bdo',
+  'br', 'Br',
+  'cite', 'Cite',
+  'code', 'Code',
+  'data', 'Data',
+  'dfn', 'Dfn',
+  'em', 'Em',
+  'i', 'I',
+  'kbd', 'Kbd',
+  'mark', 'Mark',
+  'q', 'Q',
+  'rp', 'Rp',
+  'rt', 'Rt',
+  'rtc', 'Rtc',
+  'ruby', 'Ruby',
+  's', 'S',
+  'samp', 'Samp',
+  'small', 'Small',
+  'span', 'Span',
+  'strong', 'Strong',
+  'sub', 'Sub',
+  'sup', 'Sup',
+  'time', 'Time',
+  'u', 'U',
+  'var', 'Var',
+  'wbr', 'Wbr',
+  'area', 'Area',
+  'audio', 'Audio',
+  'map', 'Map',
+  'track', 'Track',
+  'video', 'Video',
+  'embed', 'Embed',
+  'object', 'Object',
+  'param', 'Param',
+  'source', 'Source',
+  'canvas', 'Canvas',
+  'script', 'Script',
+  'noscript', 'Noscript',
+  'del', 'Del',
+  'ins', 'Ins',
+  'caption', 'Caption',
+  'col', 'Col',
+  'colgroup', 'Colgroup',
+  'table', 'Table',
+  'thead', 'Thead',
+  'tbody', 'Tbody',
+  'tfoot', 'Tfoot',
+  'td', 'Td',
+  'th', 'Th',
+  'tr', 'Tr',
+  'button', 'Button',
+  'datalist', 'Datalist',
+  'fieldset', 'Fieldset',
+  'form', 'Form',
+  'input', 'Input',
+  'label', 'Label',
+  'legend', 'Legend',
+  'meter', 'Meter',
+  'optgroup', 'Optgroup',
+  'option', 'Option',
+  'output', 'Output',
+  'progress', 'Progress',
+  'select', 'Select',
+  'textarea', 'Textarea',
+  'details', 'Details',
+  'dialog', 'Dialog',
+  'menu', 'Menu',
+  'menuitem', 'menuitem',
+  'summary', 'Summary',
+  'content', 'Content',
+  'element', 'Element',
+  'shadow', 'Shadow',
+  'template', 'Template',
+  'slot', 'Slot',
```

Comment:
Shouldn't this conflict with [slots](https://vuejs.org/v2/guide/components-slots.html)?

Reply:
Yes, same for `slot`, I've added a test to validate.